### DISCUSSION TITLE
Keep getting error code 401

### DISCUSSION THREAD

**Discussion by MahoMuri** [plhery/node-twitter-api-v2#109] [Q&A] [answered]
Hi just wanted to ask why I keep getting error code 401 for my backend requests. I used bearer token for authentication. 

Code Sample: 
```ts
import { TwitterApi } from "twitter-api-v2";

import Endpoint from "../interface/Endpoint";

const SampleEndpoint: Endpoint = {
    path: "/twitter/timeline",
    async responder(req, res, next) {
        const twitterClient = new TwitterApi(process.env.TWITTER_BEARER_TOKEN!);

        // const roClient = twitterClient.readOnly;

        const user = await twitterClient.v2.userByUsername("someuser");
        console.log(user);
    },
};

export default SampleEndpoint;
```

Am I doing it right?? Are there any problems to my code? Any form of help is much appreciated! Thank you!!

**Comment by alkihis** [comment, marked as answer]
Hello,

In theory, your code is correct.
Are you sure your bearer token is right and your ENV variable is correctly initialized?

Can you copy/paste the thrown error? 

If you want more request info, you can add, at the beginning of your script:
```ts
import { TwitterApiV2Settings } from 'twitter-api-v2';
TwitterApiV2Settings.debug = true
```
It will print sent headers, authentification and request data.

**Reply by MahoMuri** [reply]
Yup! I solved it 10 mins after I posted this. My dumb ass forgot to initialize `dotenv` in my project HAHAHA 

Sorry for wasting your time ...
